src/ppom-controller.test.ts

```diff
@@ -187,99 +249,303 @@ describe('PPOMController', () => {
+        await ppomController.updatePPOM();
+        jest.runOnlyPendingTimers();
+        expect(spy).toHaveBeenCalledTimes(5);
+      });
+
+      it('should decrease scheduleInterval is its set very high', async () => {
+        // here fileScheduleInterval is set very high but advance it by just REFRESH_TIME_DURATION
+        // is helping fetch new files as value of fileScheduleInterval is adjusted to be able to fetch all data files
+        const spy = buildFetchSpy();
+        ppomController = buildPPOMController({
+          fileScheduleInterval: REFRESH_TIME_DURATION * 100,
+        });
+        expect(spy).toHaveBeenCalledTimes(0);
+        jest.advanceTimersByTime(REFRESH_TIME_DURATION);
+        await flushPromises();
+        expect(spy).toHaveBeenCalledTimes(2);
+
+        jest.advanceTimersByTime(REFRESH_TIME_DURATION);
+        await flushPromises();
+        expect(spy).toHaveBeenCalledTimes(5);
       });
-      expect(ppomController.state.refreshInterval).toBe(0);
-      expect(spy).toHaveBeenCalledTimes(6);
     });
   });
 
   describe('clear', () => {
     it('should clear controller state', async () => {
+      buildFetchSpy();
       ppomController = buildPPOMController();
-      const spy = buildFetchSpy();
-      await ppomController.usePPOM(async () => {
-        return Promise.resolve();
-      });
-      expect(spy).toHaveBeenCalledTimes(4);
+
+      jest.runOnlyPendingTimers();
+      await ppomController.updatePPOM(false);
+
       expect(ppomController.state.storageMetadata).toHaveLength(2);
       ppomController.clear();
-      await ppomController.usePPOM(async () => {
-        return Promise.resolve();
+      expect(ppomController.state.storageMetadata).toHaveLength(0);
+    });
+  });
+
+  describe('onNetworkChange', () => {
+    it('should add network to chainIdCache if not already added', () => {
+      buildFetchSpy();
+      let callBack: any;
+      ppomController = buildPPOMController({
+        onNetworkChange: (func: any) => {
+          callBack = func;
+        },
+      });
+
+      const chainIdData1 = ppomController.state.chainIdCache.find(
+        ({ chainId }: any) => chainId === '0x1',
+      );
+      expect(chainIdData1).toBeDefined();
+      callBack({ providerConfig: { chainId: '0x2' } });
+      const chainIdData2 = ppomController.state.chainIdCache.find(
+        ({ chainId }: any) => chainId === '0x2',
+      );
+      expect(chainIdData2).toBeDefined();
+    });
+
+    it('should update lastVisited time in chainIdCache if network is already added', async () => {
+      buildFetchSpy();
+      let callBack: any;
+      ppomController = buildPPOMController({
+        onNetworkChange: (func: any) => {
+          callBack = func;
+        },
+      });
+
+      const lastVisitedBefore = ppomController.state.chainIdCache.find(
+        ({ chainId }: any) => chainId === '0x1',
+      ).lastVisited;
+
+      jest.useRealTimers();
+      await delay(10);
```

Comment:
I would try and avoid real delays in tests whenever possible for the sake of performance. Can we not replicate the same behaviour with mock timers or mocking the current date?

Reply:
I updated the PR